File: app/src/main/java/es/etologic/mahjongscoring2/data/local_data_source/local/converters/CombinationDescriptionTypeConverter.java
package es.etologic.mahjongscoring2.data.local_data_source.local.converters;

import android.arch.persistence.room.TypeConverter;

import es.etologic.mahjongscoring2.domain.entities.Combination.CombinationDescriptionType;

import static es.etologic.mahjongscoring2.domain.entities.Combination.CombinationDescriptionType.DESCRIPTION;
import static es.etologic.mahjongscoring2.domain.entities.Combination.CombinationDescriptionType.IMAGE;

public class CombinationDescriptionTypeConverter {

    @TypeConverter
    public static CombinationDescriptionType toCombinationDescriptionType(int code) {
        switch(code) {
            case 0:
                return IMAGE;
            default:
            case 1:
                return DESCRIPTION;
        }
    }

    @TypeConverter
    public static int toIntCode(CombinationDescriptionType combinationDescriptionType) {
        return combinationDescriptionType.getCode();
    }}

File: app/src/main/java/es/etologic/mahjongscoring2/data/local_data_source/local/converters/CombinationDescriptionTypeConverter.kt
package es.etologic.mahjongscoring2.data.local_data_source.local.converters

import androidx.room.TypeConverter
import es.etologic.mahjongscoring2.domain.model.Combination.CombinationDescriptionType
import es.etologic.mahjongscoring2.domain.model.Combination.CombinationDescriptionType.DESCRIPTION
import es.etologic.mahjongscoring2.domain.model.Combination.CombinationDescriptionType.IMAGE

class CombinationDescriptionTypeConverter {
    
    companion object {
        
        @TypeConverter
        @JvmStatic
        fun toCombinationDescriptionType(code: Int): CombinationDescriptionType {
            return when (code) {
                0 -> IMAGE
                else -> DESCRIPTION
            }
        }
        
        @TypeConverter
        @JvmStatic
        fun toIntCode(combinationDescriptionType: CombinationDescriptionType): Int {
            return combinationDescriptionType.code
        }
        
    }
}
